pkg/dockerregistry/server/prune/prune.go

```diff
@@ -84,8 +129,21 @@ func Prune(ctx context.Context, storageDriver driver.StorageDriver, registry dis
 
 	var stats Summary
 
-	var reposToDelete []string
+	layersPruner := &repoLayersPruner{}
+	repoPruner := &repoPruner{
+		storageDriver: storageDriver,
+	}
+
 	err = repositoryEnumerator.Enumerate(ctx, func(repoName string) error {
+		if !dryRun {
```

Comment:
This dryRun is confusing, after Enumerate you call these functions unconditionally.

Reply:
i think it's ok because the pruner's never get values set if it's dryrun.

but i agree that some comments would go a long way in this logic.